docs/architecture/search_improvements.md

```diff
@@ -0,0 +1,158 @@
+import FeatureAvailability from '@site/src/components/FeatureAvailability';
+
+# About DataHub [Search Improvements]
+
+<FeatureAvailability/>
+
+This feature adds features to our current search implementation in an effort to make search results more relevant. Included improvements are:
```

Comment:
Is this a feature? Or just... a migrating to new sarch doc?? 

We will continually make search improvements over time, so I'd suggest not naming it "Search Improvements"

Reply:
It's both, it's an overview of the changes that required the reindex and how the reindex impacts users with the additional configurations documented. "Search Improvements" was the naming we used in the town hall so that was what I went with, happy to rename it if you have another suggestion. Maybe just focus on the net new features and call it Stemming & Synonym Support?

Reply:
Yes I like that. But feel free to diverge on the format - since this really isn't a feature guide. This is more of an "upgrading DataHub guide". 

If we need 2 separate docs, 1) to discuss features and 2) to discuss the upgrade steps required, feel free to split in 2.